### ISSUE TITLE
z_loss_weight 默认是0，给出的finetune示例也是0.所以实际没有用到z loss

### ISSUE THREAD

**Issue by uygnef** [baichuan-inc/Baichuan2#156]

**Comment by mmmans** [comment]
z-loss was adopted in our training.  But it is necessary so we turn it off in the opensource code.

**Comment by uygnef** [comment]
> z-loss was adopted in our training. But it is necessary so we turn it off in the opensource code.

hi @mmmans , do you mean it's **unnecessary**  in finetune stage?

**Comment by mmmans** [comment]
> > z-loss was adopted in our training. But it is necessary so we turn it off in the opensource code.
> 
> hi @mmmans , do you mean it's **unnecessary** at finetune stage?

Yes,  not necessary. depends on your setting.

**Comment by uygnef** [comment]
oh, I see. thx a lot

**Comment by felixfuu** [comment]
@mmmans I have added thousands of new tokens and made finetuning of full parameters. Do I need to set z_loss_weight?

**Comment by mmmans** [comment]
> @mmmans I have added thousands of new tokens and made finetuning of full parameters. Do I need to set z_loss_weight?

depends on your own setting actually.  if your training does not exhibit the training instability problem, there is no need to set z_loss

**Comment by felixfuu** [comment]
@mmmans thx~

**Comment by felixfuu** [comment]
@mmmans  loss = 6.x  does not converge,should set z_loss_weight?
